packages/api/src/command/webhook/webhook.ts

```diff
@@ -80,12 +86,18 @@ export const processRequest = async (
     });
   };
 
-  const payload = webhookRequest.payload;
+  const payload = webhookRequest.payload as Record<string, string>;
   try {
+    console.log("wh key", webhookKey);
+    console.log("payload", JSON.stringify(payload.patients));
     const meta: WebhookMetadataPayload = {
       messageId: webhookRequest.id,
       when: dayjs(webhookRequest.createdAt).toISOString(),
       type: webhookRequest.type,
+      hmac: crypto
+        .createHmac("sha256", webhookKey)
+        .update(JSON.stringify(payload.patients))
```

Comment:
hashing payload.patients because customers will be hashing req.body.patients. If we hashed all of payload they would have a tougher time since they would have to remota meta from req.body and then hash req.body

Reply:
There's a weird word on your phrase, but if I understand correctly you're saying we don't want to send `meta` as part of the HMAC signature, right?

I think there's a security flaw in this approach: because the metadata is not part of the signature, an actor could intercept the message and change the `when` field.

I recommend we:
- don't send `hmac` on the body ([mentioned this on slack before](https://metriport.slack.com/archives/C04DMKE9DME/p1699026901305419?thread_ts=1699025959.407729&cid=C04DMKE9DME))
- calculate the signature/hmac using the whole WH payload, including `meta`